api/transit.go

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2025 OpenBao a Series of LF Projects, LLC
+// SPDX-License-Identifier: MPL-2.0
+
+package api
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+const (
+	keyPath       = "%s/keys/%s"
+	rotateKeyPath = "%s/keys/%s/rotate"
+
+	wrappingKeyPath = "%s/wrapping_key"
+	rewrapKeyPath   = "%s/rewrap/%s"
+
+	encryptKeyPath = "%s/encrypt/%s"
+	decryptKeyPath = "%s/decrypt/%s"
+)
+
+const (
+	plaintextKey  = "plaintext"
+	ciphertextKey = "ciphertext"
+)
+
+type Transit struct {
+	c         *Client
+	mountPath string
+}
+
+func (c *Client) Transit(mountPath string) *Transit {
+	return &Transit{
+		c:         c,
+		mountPath: mountPath,
+	}
+}
+
+func (t *Transit) CreateKeyWithContext(ctx context.Context, keyName string) error {
```

Comment:
I'd probably make a few fields required (mirroring the API itself) and then use Go's options pattern for the rest:

https://golang.design/research/generic-option/

Here, we'd probably use the actual request body as the configured struct. My 2c.